util/reggen/reg_block.py

```diff
@@ -579,6 +580,149 @@ def make_alert_regs(self, alerts: List[Alert]) -> None:
+                                             reg_width=self._reg_width,
+                                             params=self._params,
+                                             hwext=False,
+                                             default_hwqe=False,
+                                             shadowed=False,
+                                             is_alias=False,
+                                             raw=field))
+
+        reg = Register(self.offset,
+                       reg_name,
+                       None,  # no alias target
+                       reg_desc,
+                       async_name="",
+                       async_clk=None,
+                       sync_name="",
+                       sync_clk=None,
+                       hwext=False,
+                       hwqe=False,
+                       hwre=False,
+                       regwen=None,
+                       tags=[],
+                       resval=None,
+                       shadowed=False,
+                       fields=field_objs,
+                       update_err_alert=None,
+                       storage_err_alert=None)
+
+        self.add_register(reg)
+
+    def make_id_regs(self,
+                     cip_id: int,
+                     version: Version,
+                     where: str) -> None:
+        '''Insert CIP_ID and VERSION registers
+
+        The CIP_ID is a 32bit integer, and the VERSION a semantic version
+        object. The semantic version format x.y.z(+res#) is broken up into four
+        subsequent registers with values x, y, z and #. The "res#" tag is
+        optional and may be omitted.
+        '''
+
+        reg_name = "CIP_ID"
+        reg_desc = "Comportable IP ID."
+        reg_fields = [
+            {"bits": "31:0",
+             "name": "CIP_ID",
+             "desc": "This value is a unique comportable IP identifier.",
+             "resval": cip_id},
+        ]
+        self.make_ro_regs(reg_name, reg_desc, reg_fields)
+
+        # parse out build tags of the form "+res#""
+        reserved_version = 0
+        for build in version.build:
+            if re.match('res[0-9]+', build):
+                reserved_version = int(build[3:])
+
+        reg_name = "REVISION"
+        reg_desc = "Comportable IP semantic version."
+        reg_fields = [
+            {"bits": "31:24",
+             "name": "MAJOR",
+             "desc": "Major version number.",
+             "resval": version.major},
+            {"bits": "23:16",
+             "name": "MINOR",
+             "desc": "Minor version number.",
+             "resval": version.minor},
+            {"bits": "15:8",
+             "name": "SUBMINOR",
+             "desc": "Subminor (patch) version number.",
+             "resval": version.patch},
+            {"bits": "7:0",
+             "name": "RESERVED",
+             "desc": "Reserved version number.",
+             "resval": reserved_version},
+        ]
```

Comment:
I initially added four 32bit registers as outlined in the RFC, but then compacted this to one 32bit register as it felt overly excessive compared to other revision registers that we have in the system (like the [chip hardware revision registers](https://github.com/lowRISC/opentitan/blob/1f45a46e609ee3eaa8108c6fbc03888a1cf148c2/hw/ip/lc_ctrl/data/lc_ctrl.hjson#L1156-L1228) which are 16bit or 8bit). This will now look like this:

![image](https://github.com/lowRISC/opentitan/assets/41358501/6dba251f-1bcd-429e-a075-a4aa06cf2df5)

Let me know if you think this should be reverted to 4x32bit.

Reply:
If we plan to increment these sequentially I guess 8-bit fields as you have here would suffice. Wider fields can be useful if we want to use a different encoding for hardening purposes. We would probably want that only for just a handful of security-critical peripherals, I _think_. @moidx @cfrantz  any thoughts?

Reply:
I prefer using 4 8-bit fields over 4 32-bit regs for the revision. :+1: 
Your proposals for additional hardening sound reasonable to me @msfschaffner . Changing the encoding to include redundancy could mean an incompatible version change anyway. So, if we are sure we really need this, we should probably define the exact redundancy format now. But I think what @msfschaffner proposes should be sufficient.

Reply:
Discussed in the TC today - we'll go with the more compact encoding and 8bit fields.